### ISSUE TITLE
Can't connect with link-local IPv6 address

### ISSUE THREAD

**Issue by treyharris** [blinksh/blink#38]
Neither the `mosh` nor `ssh` commands will accept a link-local (`fe80::`) IPv6 address. The message returned is `Could not connect to host: Error establishing SSH session.` 

This is true whether you give the unadorned address or the `fe80::...%en0` interface-specifying one. (I've verified that the iPad has a link-local address on en0 and is addressable, e.g. with `ping6 my-ipad.local`.)

This is with today's v 0.722 on an iPad Pro 9.7 running iOS 9.3.2.

**Comment by carloscabanero** [comment]
Thanks for your report Trey! In theory, and although we use the POSIX functions, it should be IPv6 friendly. I really didn't test this case but it is really important. Would you mind running with --verbose to see exactly where it fails? Thanks again!

**Comment by treyharris** [comment]
(I've obscured the address below.)

```
blink> mosh trey@fe80::... --verbose
MoshClient:DEBUG:mosh-server new -s -c 256 -l LC_ALL=en_US.UTF-8 -- 
MoshClient:DEBUG:ssh -v -t trey@fe80::... -- mosh-server new -s -c 256 -l LC_ALL=en_US.UTF-8 -- 
SSHSession:DEBUG:Failed sending banner
Could not connect to host: Error establishing SSH session.
Did not find remote IP address
```

Repeating with `...%en0` made no difference. That said, though, mosh itself doesn't support IPv6 addresses yet (see mobile-shell/mosh#81), so I wouldn't expect this to work. I _would_ expect ssh to work, though, as ssh has had IPv6 support for years. But that, too, fails with the `SSHSession:DEBUG:Failed sending banner` error. Here's a sample where I attempt to login to the same machine via ssh with

1. the `.local` hostname
1. the IPv4 address
1. the IPv6 address _with_ `%en0`
1. the IPv6 address _without_ `%en0`

```
blink> ssh -v trey@trey-laptop.local
SSHSession:DEBUG:Connecting to trey-laptop.local [10.75.108.174] port 22.
Connected to 10.75.108.174
SSHSession:DEBUG:Authenticating as 'trey'.
SSHSession:DEBUG:Attempting authentication with publickey.
SSHSession:DEBUG:Authentication succeeded.
SSHSession:DEBUG:ssh_session_start: channel created
SSHSession:DEBUG:ssh_session_start: pty requested
SSHSession:DEBUG:requesting shell
SSHSession:DEBUG:shell request accepted
Last login: Fri Jun 24 14:09:21 2016 from fe80::f65c:89ff:feab:d3e9%en0
% exit
SSHSession:DEBUG:session finished with code 0
blink> ssh -v trey@10.75.108.174
SSHSession:DEBUG:Connecting to 10.75.108.174 [10.75.108.174] port 22.
Connected to 10.75.108.174
The authenticity of host 10.75.108.174 (10.75.108.174) can't be established.
RSA key fingerprint is [REDACTED]
Are you sure you want to continue connecting (yes/no)?yes
Permanently added key for 10.75.108.174 to list of known hosts.
SSHSession:DEBUG:Authenticating as 'trey'.
SSHSession:DEBUG:Attempting authentication with publickey.
SSHSession:DEBUG:Authentication succeeded.
SSHSession:DEBUG:ssh_session_start: channel created
SSHSession:DEBUG:ssh_session_start: pty requested
SSHSession:DEBUG:requesting shell
SSHSession:DEBUG:shell request accepted
Last login: Fri Jun 24 14:09:30 2016 from 10.75.108.157
% ifconfig en0 | perl -lne 'print $1 if m/(fe80::\S+)/'
fe80::[REDACTED]:a526%en0
% exit
SSHSession:DEBUG:session finished with code 0
blink> ssh -v trey@fe80::[REDACTED]:a526%en0
SSHSession:DEBUG:Failed sending banner
Could not connect to host: Error establishing SSH session.
blink> ssh -v trey@fe80::[REDACTED]:a526
SSHSession:DEBUG:Failed sending banner
Could not connect to host: Error establishing SSH session.
```

For comparison, here's connecting to the same host from Mac OS X El Capitan:
```
% ssh -v trey@fe80::[REDACTED]:a526%en
OpenSSH_6.9p1, LibreSSL 2.1.8
debug1: Reading configuration data /Users/trey/.ssh/config
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 21: Applying options for *
debug1: Connecting to fe80::[REDACTED]:a526%en0 [fe80::[REDACTED]:a526%en0] port 22.
debug1: Connection established.
debug1: identity file /Users/trey/.ssh/id_ed25519-cert type -1
debug1: Enabling compatibility mode for protocol 2.0
debug1: Local version string SSH-2.0-OpenSSH_6.9
debug1: Remote protocol version 2.0, remote software version OpenSSH_6.9
debug1: match: OpenSSH_6.9 pat OpenSSH* compat 0x04000000
debug1: Authenticating to fe80::[REDACTED]:a526%en0:22 as 'trey'
debug1: SSH2_MSG_KEXINIT sent
debug1: SSH2_MSG_KEXINIT received
debug1: kex: server->client chacha20-poly1305@openssh.com <implicit> none
debug1: kex: client->server chacha20-poly1305@openssh.com <implicit> none
debug1: expecting SSH2_MSG_KEX_ECDH_REPLY
debug1: Server host key: ecdsa-sha2-nistp256 SHA256:/[REDACTED]
debug1: Host 'fe80::[REDACTED]:a526%en0' is known and matches the ECDSA host key.
debug1: Found key in /Users/trey/.ssh/known_hosts:3
debug1: SSH2_MSG_NEWKEYS sent
debug1: expecting SSH2_MSG_NEWKEYS
debug1: SSH2_MSG_NEWKEYS received
debug1: SSH2_MSG_SERVICE_REQUEST sent
debug1: SSH2_MSG_SERVICE_ACCEPT received
debug1: Authentications that can continue: publickey,keyboard-interactive
debug1: Next authentication method: publickey
debug1: Offering RSA public key: /Users/trey/.ssh/id_rsa
debug1: Server accepts key: pkalg ssh-rsa blen 279
debug1: Authentication succeeded (publickey).
Authenticated to fe80::[REDACTED]:a526%en0 ([fe80::[REDACTED]:a526%en0]:22).
debug1: channel 0: new [client-session]
debug1: Requesting no-more-sessions@openssh.com
debug1: Entering interactive session.
debug1: client_input_global_request: rtype hostkeys-00@openssh.com want_reply 0
debug1: Sending environment.
debug1: Sending env LANG = en_US.UTF-8
Last login: Fri Jun 24 14:20:28 2016 from @fe80::[REDACTED]:a526%en0
% exit
debug1: client_input_channel_req: channel 0 rtype exit-status reply 0
debug1: client_input_channel_req: channel 0 rtype eow@openssh.com reply 0
debug1: channel 0: free: client-session, nchannels 1
Connection to fe80::[REDACTED]:a526%en0 closed.
Transferred: sent 3712, received 4844 bytes, in 6.8 seconds
Bytes per second: sent 546.7, received 713.4
debug1: Exit status 1
```

**Comment by carloscabanero** [comment]
This should already be fixed on v3.0.21 :)

Let me know if you have any issues!
